### PR TITLE
esn-frontend-mailto#13: Added a 'cancelDestroy' method to InboxDraft that allows cancelling the destruction of a draft

### PR DESCRIPTION
**Pull request by huy-ta**
Partially resolves https://github.com/OpenPaaS-Suite/esn-frontend-mailto/issues/13.

- TODO:

  - [x] Writing tests

### CODE REVIEW

**Review by chamerling — APPROVED** [review]
LGTM

**Review by piavgh — APPROVED** [review]

**Review by rezk2ll — APPROVED** [review]

File: src/linagora.esn.unifiedinbox/app/components/composer/composer.controller.js [hunk 1]
```diff
@@ -168,6 +168,15 @@ angular.module('linagora.esn.unifiedinbox')
       _.forEach(self.message.attachments, _cancelAttachment);
 
       self.draft.destroy(options).then(self.onDiscard, self.onShow);
+
+      if (typeof self.onDiscarding !== 'function') return;
```
Comment (tuanlc) [inline src/linagora.esn.unifiedinbox/app/components/composer/composer.controller.js:172]: Missing test for this?

**Review by tuanlc — APPROVED** [review]

File: src/linagora.esn.unifiedinbox/app/components/composer/composer.controller.js (hunk 1 above)
Comment (huy-ta) [reply src/linagora.esn.unifiedinbox/app/components/composer/composer.controller.js:172]: It's just a simple guard check so it doesn't really make sense to write a test for it.

Comment (tuanlc) [reply src/linagora.esn.unifiedinbox/app/components/composer/composer.controller.js:172]: Ok

**Review by rezk2ll — APPROVED** [review]

**Pull request merged**